Sources/Xcodeproj/Plist.swift

```diff
@@ -0,0 +1,68 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+/// A enum representing data types for legacy Plist type.
+/// see: https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/PropertyLists/OldStylePlists/OldStylePLists.html
+enum Plist {
+    case string(String)
+    case array([Plist])
+    case dictionary([String: Plist])
+}
+
+extension Plist: ExpressibleByStringLiteral {
+    public typealias UnicodeScalarLiteralType = StringLiteralType
+    public typealias ExtendedGraphemeClusterLiteralType = StringLiteralType
+
+    public init(unicodeScalarLiteral value: UnicodeScalarLiteralType) {
+        self = .string(value)
+    }
+    public init(extendedGraphemeClusterLiteral value: ExtendedGraphemeClusterLiteralType) {
+        self = .string(value)
+    }
+    public init(stringLiteral value: StringLiteralType) {
+        self = .string(value)
+    }
+}
+
+private let escapedQuote = "\\\""
+
+extension Plist {
+    /// Serializes the Plist enum to string.
+    func serialize() -> String {
+        switch self {
+        case .string(let str):
+            return "\"" + escape(string: str) + "\""
+        case .array(let items):
+            return "(" + items.map { $0.serialize() }.joined(separator: ", ") + ")"
+        case .dictionary(let items):
+            return "{" + items.map { " \($0) = \($1.serialize()) " }.joined(separator: "; ") + "; };"
+        }
+    }
+
+    /// Escapes the string for plist.
+    /// Finds the instances of quote (") and backward slash (\) and prepends
+    /// the escape character backward slash (\).
+    private func escape(string: String) -> String {
+        guard let pos = string.utf8.index(where: { $0 == UInt8(ascii: "\\") || $0 == UInt8(ascii: "\"") }) else {
+            return string
+        }
+        var newString = String(string.utf8[string.utf8.startIndex..<pos])!
+        for char in string.utf8[pos..<string.utf8.endIndex] {
+            switch char {
```

Comment:
Might worth combining this with the `$0 == UInt8(ascii: "\\") || $0 == UInt8(ascii: "\"")` above into a simple helper closure or function called `needsEscape()` or something like that.  The compiler should be able to inline it, and it both simplifies the code and makes it really clear what's going on.  Just a thought...